### PR TITLE
upgradeseries is only applicable for IAAS models

### PR DESCRIPTION
**Pull request by wallyworld**
## Description of change

When setting up the uniter remote watcher, only hook up the upgradeseries watcher on IAAS models.

As a driveby, move a couple of storage watcher vars left in the wrong place.

## QA steps

Ensure CAAS deployments work again.

### CODE REVIEW

File: worker/uniter/remotestate/watcher.go [hunk 1]
```diff
@@ -251,10 +251,10 @@ func (w *RemoteStateWatcher) loop(unitTag names.UnitTag) (err error) {
 	requiredEvents++
 
 	var (
-		seenStorageChange bool
-		storageChanges    watcher.StringsChannel
-
 		seenApplicationChange *bool
+
+		upgradeSeriesChanges    watcher.NotifyChannel
+		seenUpgradeSeriesChange *bool
```
Comment (jameinel) [inline worker/uniter/remotestate/watcher.go]: pointer to bool seems so fragile. Do we need the 'nil' state to know that it was never initialized? It seems safer to just always have the bool even if it doesn't mean anything sometimes.
I realize seenApplicationChange follows that pattern, but it just feels like an easy way to mess up and get a panic.

Comment (wallyworld) [reply worker/uniter/remotestate/watcher.go]: I've reworked it

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**